server/lib/scraping_job_schema.js

```diff
@@ -1,23 +1,24 @@
 /**
-* Exports a mongoose schema representing a facebook job
+* Exports a mongoose schema representing a scraping job
 **/
 
 import mongoose from 'mongoose';
 import Q from 'q';
 import moment from 'moment';
 
-const facebookJobSchema = new mongoose.Schema({
-  facebook_account: String,
-  facebook_token: String,
+const scrapingJobSchema = new mongoose.Schema({
+  type: String,
+  account: String,
+  token: String,
   scraping_interval: Number, // value in minutes
   next_run: Date,
 });
 
 /**
-* Finds all the facebook jobs that should be performed
+* Finds all the scraping jobs that should be performed according to current time.
 * @returns {Promise} resolving to the list of jobs
 **/
-facebookJobSchema.statics.findJobsToRun = function findJobsToRun() {
+scrapingJobSchema.statics.findJobsToRun = function findJobsToRun() {
```

Comment:
Consider the following situation:
You have your polling interval (looking for jobs to run) set to one minute. If one of your jobs is running for longer then a minute, it will be pushed again to the queue, as the sheduleNextRun function is not called until the job is finished.
This could result in the same jobs being pushed to the queue over and over again.
There are several solutions. One would be to set a field on the object indicating that it was already sent to the queue, to avoid the additional runs before we update the nextRun field. Another way would be to update nextRun field immediately after queuing the job. In case of an error, you could always override the field to try again immediately, or wait few minutes and then try again.